integration/map_test.go

```diff
@@ -31,59 +31,138 @@ import (
 	_ "github.com/google/trillian/merkle/maphasher"
 )
 
-// createHashKV returns a []*trillian.MapLeaf formed by the mapping of index, value ...
-// createHashKV panics if len(iv) is odd. Duplicate i/v pairs get over written.
-func createMapLeaves(iv ...[]byte) []*trillian.MapLeaf {
-	if len(iv)%2 != 0 {
-		panic(fmt.Sprintf("integration: createMapLeaves got odd number of iv pairs: %v", len(iv)))
+// newTreeWithHasher is a test setup helper for creating new trees with a given hasher.
+func newTreeWithHasher(ctx context.Context, env *integration.MapEnv, hashStrategy trillian.HashStrategy) (*trillian.Tree, hashers.MapHasher, error) {
+	treeParams := stestonly.MapTree
+	treeParams.HashStrategy = hashStrategy
+	tree, err := env.AdminClient.CreateTree(ctx, &trillian.CreateTreeRequest{
+		Tree: treeParams,
+	})
+	if err != nil {
+		return nil, nil, nil
 	}
-	r := []*trillian.MapLeaf{}
-	for i := 0; i < len(iv); i += 2 {
-		r = append(r, &trillian.MapLeaf{
-			Index:     iv[i],
-			LeafValue: iv[i+1],
-		})
+
+	hasher, err := hashers.NewMapHasher(tree.HashStrategy)
+	if err != nil {
+		return nil, nil, nil
 	}
-	return r
+	return tree, hasher, nil
 }
 
-func TestInclusionWithEnv(t *testing.T) {
+func TestInclusion(t *testing.T) {
 	ctx := context.Background()
-	env, err := integration.NewMapEnv(ctx, "TestInclusionWithEnv")
+	env, err := integration.NewMapEnv(ctx, "TestInclusion")
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer env.Close()
 	for _, tc := range []struct {
 		desc string
 		trillian.HashStrategy
-		index, value []byte
+		iv [][]byte
 	}{
 		{
-			desc:         "maphasher",
+			desc:         "maphasher single",
+			HashStrategy: trillian.HashStrategy_TEST_MAP_HASHER,
+			iv: [][]byte{
+				testonly.TransparentHash("A"), []byte("A"),
+			},
+		},
+		{
+			desc:         "maphasher multi",
 			HashStrategy: trillian.HashStrategy_TEST_MAP_HASHER,
-			index:        testonly.TransparentHash("A"),
-			value:        []byte("A"),
+			iv: [][]byte{
+				testonly.TransparentHash("A"), []byte("A"),
+				testonly.TransparentHash("B"), []byte("B"),
+				testonly.TransparentHash("C"), []byte("C"),
+			},
 		},
 	} {
-		treeParams := stestonly.MapTree
-		treeParams.HashStrategy = tc.HashStrategy
-		tree, err := env.AdminClient.CreateTree(ctx, &trillian.CreateTreeRequest{
-			Tree: treeParams,
-		})
+		tree, hasher, err := newTreeWithHasher(ctx, env, tc.HashStrategy)
 		if err != nil {
-			t.Errorf("%v: CreateTree(): %v", tc.desc, err)
+			t.Errorf("%v: newTreeWithHasher(%v): %v", tc.desc, tc.HashStrategy, err)
```

Comment:
Yep.  I removed the `Large` test since it's really a duplicate of the `Batch` test.
The batch test has proper continues.